### PR TITLE
Exploratory commit [DO NOT MERGE!]

### PR DESCRIPTION
**Pull request by radulucaciu**
This is an exploratory PR, for tutoring purposes only.

### CODE REVIEW

**Comment by radulucaciu** [comment]
@jozsi from what i understand, components are used to break up presentation items in manageable chunks. In this case I created a component to display a simple message when the user does not have any wallets. 
I tried:
 - creating a new component (called NoData, please ignore the Wallets component for now)
 - including the newly created component in the Home container.
I have a few questions addressed as comments below

File: src/containers/Home.jsx [hunk 1]
```diff
@@ -57,6 +58,9 @@ class Home extends Component {
               })}
             />
           </Box>
+          if (wallet.length === 0) {
```
Comment (radulucaciu) [inline src/containers/Home.jsx:61]: How can we, if at all, add conditions inside an existing component?

File: src/containers/Home.jsx [hunk 2]
```diff
@@ -57,6 +58,9 @@ class Home extends Component {
...
+            <NoData />
+          }
           <List selectable>
```
Comment (radulucaciu) [inline src/containers/Home.jsx:64]: I am thinking of moving this into its own component so that it can be included in the conditional above

**Comment by jozsi** [comment]
Let me explain a few concepts very briefly:
- (presentational) components represent the look
- container (components) represent the logic

The current front-end code is really experimental, so let me showcase a correct example:
**WalletsContainer**
```html
<div>
  <Header title="My Wallets" />
  <WalletsList wallets={state.wallets} />
</div>
```

`WalletsContainer` also contains the logic that loads the list of wallets from the API into the containers `state`, and passes it over to the `WalletsList` as props. It's markup just wraps components without any styling (that's why we need a `Header` component that decides styling, instead of using an `h2` tag).

**WalletsList**
```html
<div>
  { props.wallets.length
    ? props.wallets.map(wallet => <ListItem text={wallet.name} secondaryText={wallet.balance} />)
    : <ListEmpty />
  }
</div>
``` 

`WalletsList` is a stateless component, it just renders the UI for the data received via `props` from the container. The "logic" inside only refers to styling the markup. `ListEmpty` would just be a message paragraph.

[This article](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.tctgcu60y) explains the differences in depth.

Grommet fortunately provides several components, like [List](https://grommet.github.io/docs/list) and [ListPlaceholder](https://grommet.github.io/docs/list-placeholder). Because their `ListPlaceholder` has more options (loading spinner, empty message, filtering), the logic is delegated to it:
```html
<List>
  { props.wallets.map(wallet => <ListItem text={wallet.name} secondaryText={wallet.balance} />) }
  <ListPlaceholder
    emptyMessage="No wallets"
    unfilteredTotal={props.wallets.length}
    filteredTotal={props.isLoading ? undefined : props.wallets.length}
  />
</List>
```
Looking at it's [source code](https://github.com/grommet/grommet-addons/blob/master/src/js/components/ListPlaceholder.js), we see that:
- if `filteredTotal` is undefined, it will display a loading spinner
- if `unfilteredTotal` is 0, `emptyMessage` is displayed
- else, nothing is rendered (`null`)

The above code won't work out of the box, because:
- `ListPlaceholder` is part of `grommet-addons` package, that's not yet added to the project
- we don't have an `isLoading` state for the wallet, yet; however, that would just skip the loading spinner

I'll implement all of these and open up a PR, with a deep-dive into redux (describing how the `isLoading` state is implemented and hooked up with the container).

**Pull request closed without merging**